x-pack/plugins/security_solution/cypress/e2e/detection_rules/custom_query_rule.cy.ts

```diff
@@ -105,98 +133,122 @@ describe('Custom query rules', () => {
     login();
   });
   describe('Custom detection rules creation', () => {
-    const expectedUrls = getNewRule().referenceUrls?.join('');
-    const expectedFalsePositives = getNewRule().falsePositivesExamples?.join('');
-    const expectedTags = getNewRule().tags?.join('');
-    const mitreAttack = getNewRule().mitre as Mitre[];
-    const expectedMitre = formatMitreAttackDescription(mitreAttack);
     const expectedNumberOfRules = 1;
 
     beforeEach(() => {
-      const timeline = getNewRule().timeline as CompleteTimeline;
       deleteAlertsAndRules();
-      createTimeline(timeline).then((response) => {
-        cy.wrap({
-          ...getNewRule(),
-          timeline: {
-            ...timeline,
-            id: response.body.data.persistTimeline.timeline.savedObjectId,
-          },
-        }).as('rule');
-      });
+      createTimeline(getTimeline())
+        .then((response) => {
+          return response.body.data.persistTimeline.timeline.savedObjectId;
+        })
+        .as('timelineId');
     });
 
     it('Creates and enables a new rule', function () {
       visit(RULE_CREATION);
-      fillDefineCustomRuleAndContinue(this.rule);
-      fillAboutRuleAndContinue(this.rule);
-      fillScheduleRuleAndContinue(this.rule);
+
+      cy.log('Filling define section');
+      importSavedQuery(this.timelineId);
+      continueWithNextSection();
+
+      cy.log('Filling about section');
+      fillRuleName();
+      fillDescription();
+      fillSeverity();
+      fillRiskScore();
+      fillRuleTags();
+      expandAdvancedSettings();
+      fillReferenceUrls();
+      fillFalsePositiveExamples();
+      fillThreat();
+      fillThreatTechnique();
+      fillThreatSubtechnique();
+      fillNote();
+      continueWithNextSection();
+
+      cy.log('Filling schedule section');
+      fillFrom();
 
       // expect define step to repopulate
       cy.get(DEFINE_EDIT_BUTTON).click();
-      cy.get(CUSTOM_QUERY_INPUT).should('have.value', this.rule.customQuery);
+      cy.get(CUSTOM_QUERY_INPUT).should('have.value', getQuery());
       cy.get(DEFINE_CONTINUE_BUTTON).should('exist').click({ force: true });
       cy.get(DEFINE_CONTINUE_BUTTON).should('not.exist');
 
       // expect about step to populate
       cy.get(ABOUT_EDIT_BUTTON).click();
-      cy.get(RULE_NAME_INPUT).invoke('val').should('eql', this.rule.name);
+      cy.get(RULE_NAME_INPUT).invoke('val').should('eql', getRuleName());
```

Comment:
Same here and other similar places in this test.

```ts
const ruleName = getRuleName();
// ...
fillRuleName(ruleName);
// ...
cy.get(RULE_NAME_INPUT).invoke('val').should('eql', ruleName);
```

Sometimes you don't really care about exact value that will be filled in. Like here, you probably case that _some_ rule name will be filled, and then whatever was filled will be shown when you switch back to the About step. In this case the code could be simplified this way:

```ts
export const fillRuleName = (ruleName: string = getRuleName()) => {
  cy.get(RULE_NAME_INPUT).clear({ force: true }).type(ruleName, { force: true });
  return ruleName;
};
```

```ts
const ruleName = fillRuleName();
// ...
cy.get(RULE_NAME_INPUT).invoke('val').should('eql', ruleName);
```